src/repositories/request-repository.ts

```diff
@@ -183,16 +192,28 @@ export class RequestRepository extends Repository<Request> {
           processingStatus: RequestStatus.PROCESSING,
           processingDeadline: DateUtils.mixedDateToUtcDatetimeString(processingDeadline),
         })
+        .orWhere(
+          'request.status = :failedStatus AND request.createdAt >= :earliestDateToRetry AND (request.message IS NULL OR request.message != :message)',
+          {
+            failedStatus: RequestStatus.FAILED,
+            earliestDateToRetry: DateUtils.mixedDateToUtcDatetimeString(earliestDateToRetry),
+            message: REQUEST_MESSAGES.conflictResolutionRejection,
+          }
+        )
         .orWhere('request.status = :pendingStatus', { pendingStatus: RequestStatus.PENDING })
         .orderBy('MIN(request.createdAt)', 'ASC')
         .groupBy('request.streamId')
+        // if 0 will return unlimited
         .limit(streamLimit)
         .getMany()
 
       // Do not anchor if the earliest request isn't expired and there isn't enough streams
       const earliestIsNotExpired =
         streamsToAnchor.length > 0 && streamsToAnchor[0].createdAt >= anchoringDeadline
-      if (earliestIsNotExpired && streamsToAnchor.length < streamLimit) {
+      if (
+        streamsToAnchor.length === 0 ||
+        (earliestIsNotExpired && streamsToAnchor.length < streamLimit)
+      ) {
```

Comment:
Can we add some logging here to make clear what is happening?

Reply:
The logic seems to be quite confusing here so I'm going to make some changes and add logging